onadata/libs/tests/utils/test_logger_tools.py

```diff
@@ -24,6 +36,68 @@ def test_generate_content_disposition_header(self):
 
         return_value_with_name_and_false_show_date = \
             generate_content_disposition_header(file_name, extension, False)
-        print return_value_with_name_and_false_show_date
         self.assertTrue(re.search(file_name_pattern,
                                   return_value_with_name_and_false_show_date))
+
+    def test_attachment_tracking(self):
+        """
+        Test that when a submission with many attachments is made,
+        we keep track of the total number of attachments expected
+        and if we have received all of them
+        """
+        md = """
+        | survey |       |        |       |
+        |        | type  | name   | label |
+        |        | image | image1 | Photo |
+        |        | image | image2 | Photo |
+        """
+        self._create_user_and_login()
+        self.xform = self._publish_md(md, self.user)
+
+        xml_string = """
+        <data id="{}">
+            <image1>1300221157303.jpg</image1>
+            <image2>1300375832136.jpg</image2>
+        </data>
+        """.format(self.xform.id_string)
+        file_path = "{}/apps/logger/tests/Health_2011_03_13."\
+                    "xml_2011-03-15_20-30-28/1300221157303"\
+                    ".jpg".format(settings.PROJECT_ROOT)
+        media_file = django_file(path=file_path,
+                                 field_name="image1",
+                                 content_type="image/jpeg")
+        instance = create_instance(self.user.username,
+                                   StringIO(xml_string.strip()),
+                                   media_files=[media_file])
+        self.assertFalse(instance.json[MEDIA_ALL_RECEIVED])
+        self.assertEquals(instance.json[TOTAL_MEDIA], 2)
+        self.assertEquals(instance.json[MEDIA_COUNT], 1)
+        self.assertEquals(instance.json[TOTAL_MEDIA], instance.total_media)
+        self.assertEquals(instance.json[MEDIA_COUNT], instance.media_count)
+        self.assertEquals(instance.json[MEDIA_ALL_RECEIVED],
+                          instance.media_all_received)
+        xml2_string = """
+        <data id="{}">
```

Comment:
You should use the same submission XML as above, you can add the `instanceID`, I see no need for the two.